cmake/mkl/MKLConfig.cmake

```diff
@@ -758,41 +850,17 @@ foreach(lib ${MKL_REQUESTED_LIBRARIES})
 endforeach()
 
 # Threading selection
-if(MKL_THREADING STREQUAL "tbb_thread" OR MKL_DPCPP_THREADING STREQUAL "tbb_thread")
-  find_package(TBB CONFIG COMPONENTS tbb )
-  if(TARGET TBB::tbb)
-    if(MKL_THREADING STREQUAL "tbb_thread")
-      set(MKL_THREAD_LIB $<TARGET_LINKER_FILE:TBB::tbb>)
-      set(MKL_SDL_THREAD_ENV "TBB")
-    endif()
-    if(MKL_DPCPP_THREADING STREQUAL "tbb_thread")
-      set(MKL_DPCPP_THREAD_LIB $<TARGET_LINKER_FILE:TBB::tbb>)
-    endif()
-    get_property(TBB_LIB TARGET TBB::tbb PROPERTY IMPORTED_LOCATION_RELEASE)
-    get_filename_component(TBB_LIB_DIR ${TBB_LIB} DIRECTORY)
-  else()
-    if(UNIX)
-      set(TBB_LIBNAME libtbb.so)
-    else()
-      set(TBB_LIBNAME tbb.lib)
-    endif()
-    find_path(TBB_LIB_DIR ${TBB_LIBNAME}
-        HINTS $ENV{TBBROOT} $ENV{MKLROOT} ${MKL_ROOT} ${TBB_ROOT}
-        PATH_SUFFIXES "lib" "lib/intel64/gcc4.4" "lib/intel64/gcc4.8"
-                 "../tbb/lib/intel64/gcc4.4" "../tbb/lib/intel64/gcc4.8"
-                 "../../tbb/latest/lib/intel64/gcc4.8"
-                 "../tbb/lib/intel64/vc14" "lib/intel64/vc14"
-    )
-    find_library(TBB_LIBRARIES NAMES tbb
-        HINTS $ENV{TBBROOT} $ENV{MKLROOT} ${MKL_ROOT} ${TBB_ROOT}
-        PATH_SUFFIXES "lib" "lib/intel64/gcc4.4" "lib/intel64/gcc4.8"
-                 "../tbb/lib/intel64/gcc4.4" "../tbb/lib/intel64/gcc4.8"
-                 "../../tbb/latest/lib/intel64/gcc4.8"
-                 "../tbb/lib/intel64/vc14" "lib/intel64/vc14"
-    )
-    include(FindPackageHandleStandardArgs)
-    find_package_handle_standard_args(MKL REQUIRED_VARS TBB_LIBRARIES)
+if(MKL_THREADING STREQUAL "tbb_thread" OR MKL_SYCL_THREADING STREQUAL "tbb_thread")
+  find_package(TBB REQUIRED CONFIG COMPONENTS tbb)
+  if(MKL_THREADING STREQUAL "tbb_thread")
+    set(MKL_THREAD_LIB $<TARGET_LINKER_FILE:TBB::tbb>)
```

Comment:
Similar to https://github.com/oneapi-src/oneMKL/pull/354#discussion_r1272658210 could you please keep the old TBB search in case we don't have TBBConfig.cmake available in the package?

Reply:
Thank you, Maria!
Updated